### PR TITLE
LDAPS, Identicons, and a happier Gemnasium

### LINKED ISSUE #173: LDAPS Implementation Crashes the server - Need to implement LDAPS again when we can
Two similar-looking issues that can occur in a variety of combinations (i.e. only issue 1, only issue 2, both issue 1 and 2):

The first issue is ldap related:
```
LDAP connection error: { [Error: read ECONNRESET] code: 'ECONNRESET', errno: 'ECONNRESET', syscall: 'read' }
```
I think I have a fix for it, though I'm not a huge fan of the fix, it might be the only way. Not sure on this yet. I was about to test the fix, but the server crashed at the same time (unrelated to my fix, see next issue).

The other issue actually crashes the server which is not good.
```
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at exports._errnoException (util.js:874:11)
    at TCP.onread (net.js:544:26)
 [16:24:24] [nodemon] app crashed - waiting for file changes before starting...
```
I noticed there was an issue logged against node that sounds like what we're dealing with: [nodejs/node#3595](https://github.com/nodejs/node/issues/3595). So I am now reverting back to `0.12.8` node.js until we understand more on this.

It appears to have something to do with idling sockets based on my observations and form what the above nodejs issue talks about.

### PR DESCRIPTION
**Pull request by Mctalian**
**Upgraded passport-windowsauth to latest version 1.0.0**
My PR for passport-windowsauth was accepted, so LDAPS is now possible again.

**Upgraded angular-identicon to latest version 0.1.18**
My PR for angular-identicon was accepted, so identicons of all sizes will look symmetric now:

image
Old is on top, new is on bottom. Subtle difference, but it bugged me forever.

**Moved non-dev dependencies out of devDependencies**
passport, passport-windowsauth, and mongodb are used outside of just dev.

**Upgraded mongodb and alphabetized bower.json**
Upgraded mongodb to latest and alphabetized bower.json for organization.

**Reviewers**
- [x] @YashdalfTheGray
- [x] @bcbrennecke 

Closes #173

### CODE REVIEW

**Comment by alanlgirard** [comment]
Finally on the identicon PR!  Just in time to scrap it all for skype for business photos! :stuck_out_tongue:

**Comment by alanlgirard** [comment]
Approved but I will allow @bcbrennecke to be the final approval

File: bower.json [hunk 1]
```diff
@@ -18,21 +18,21 @@
...
-    "lodash": "3.10.0",
-    "moment": "2.10.3",
     "event-source-polyfill": "0.0.6",
     "identicon": "*",
-    "angular-identicon": "0.1.17",
-    "angular-timeline": "1.6.1",
     "jqcloud2": "2.0.1",
-    "jquery": "2.2.0"
+    "jquery": "2.2.0",
+    "lodash": "3.10.0",
```
Comment (YashdalfTheGray) [inline bower.json]: Lodash needs the version update.

File: package.json [hunk 2]
```diff
@@ -32,7 +32,10 @@
     "external-ip": "0.2.3",
     "ip": "0.3.3",
     "lodash": "3.10.0",
```
Comment (YashdalfTheGray) [inline package.json]: Lodash needs the version update.

**Comment by Mctalian** [comment]
Check it again @YashdalfTheGray, make sure I didn't make a boo-boo

**Comment by bcbrennecke** [comment]
Geez, 10 commits for that?  No wonder you're ahead of @monotkate  and I.
Approved

**Comment by Mctalian** [comment]
haha I don't think merges count in the totals FYI! So this is more like 5

**Pull request merged**